.github/workflows/test-python.yml

```
@@ -0,0 +1,37 @@
+name: test-python
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  PYTHON_VERSION: "3.10"
+    
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          architecture: x64
+
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
```

```suggestion
        uses: PyO3/maturin-action@6c20e5e8673d2470a20e14adf067b68db6fee644 # v1.43.0
```

Pinning to hashes for workflows by non-verified authors is [recommended](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions).

---

Great suggestion!